dvclive/live.py

```diff
@@ -180,6 +183,10 @@ def make_summary(self):
         with open(self.summary_path, "w") as f:
             json.dump(summary_data, f, indent=4)
 
+    def make_report(self, mode: str = "html"):
```

Comment:
Users can manually call this at any point from a `Live` instance

Reply:
Hm, it seems odd that `report` is kwarg in `Live()`, but it doesn't get passed in `live.make_report()` calls (unlike `live.next_step()`). Seems like it should either default to whatever is passed in `Live()`, or `report` kwarg should be removed from `Live()`.

Related: It seems odd to pass `Live(report="html")` in no-step scenarios and have that kwarg do nothing. So maybe that's not the best place for it? Not sure 🤔 .

Reply:
I'm not sure either.

It might be better to remove `mode` from `make_report` and make the `report` arg of `Live` the only source of truth.

One problem with the alternative (removing `report` from `Live`) is that integrations become a little more complicated because right now users can customize `Live` args but in this case we would need to handle `report` separately

Reply:
> One problem with the alternative (removing `report` from `Live`) is that integrations become a little more complicated because right now users can customize `Live` args but in this case we would need to handle `report` separately

Got it, makes sense.

> It might be better to remove `mode` from `make_report` and make the `report` arg of `Live` the only source of truth.

Since it's a method of `Live`, I would expect it to use `self._report` unless I override it, so what about something like this?

```
    def make_report(self, mode: Optional[str] = None):
           if not mode:
               mode = self._report
```

Also, maybe we should rename in one place or the other so it is consistently either `report` or `mode`?

Reply:
> It seems odd to pass Live(report="html") in no-step scenarios and have that kwarg do nothing. 

Not sure if it should be part of this PR, but currently, I can use the `live` section in `dvc.yaml` to get a report even in no-step scenarios, but it doesn't seem like it's possible without the `live` section. Is there some way to still generate the report by default?

Reply:
>  Is there some way to still generate the report by default?

When using the `live` section in no-step scenario, the report gets generated by the monitor task after the stage has ended. 
Any implementation on DVCLive side would behave differently, as the `Live` object doesn't have info of when the "stage" has ended (the closest would be to implement the logic in `__del__`) .
IMO would be better to document `Live.make_report()` and emphasize it's usage for non step scenarios

Reply:
Extracted to #231